R/simulation-batch.R

```diff
@@ -68,29 +69,62 @@ SimulationBatch <- R6::R6Class(
         stop(messages$errorOnlyOneValuesSetAllowed("parameterValues, initialValues"))
       }
 
+      # Check if any of the values is `NA`. If so, throw an error, as such
+      # values set will produce empty results
+      if (any(is.na(parameterValues))) {
+        naIdx <- which(is.na(parameterValues))
+        stop(messages$simBatchStartValueNaN(self$getVariableParameters()[naIdx]))
+      }
+      if (any(is.na(initialValues))) {
+        naIdx <- which(is.na(initialValues))
+        stop(messages$simBatchStartValueNaN(self$getVariableMolecules()[naIdx]))
+      }
+
       batchRunValues <- SimulationBatchRunValues$new(parameterValues, initialValues)
       rClr::clrCall(self$ref, "AddSimulationBatchRunValues", batchRunValues$ref)
       return(batchRunValues$id)
     },
 
+    #' @description Returns a list of parameter paths that are variable in this batch.
+    #' @details The order of parameters is the same as the order of parameter
+    #' values added with `$addRunValues()` method.
+    #'
+    #' @return List of parameter paths, or `NULL` if no parameter is variable.
+    #' @export
+    getVariableParameters = function() {
+      simulationBatchOptions <- rClr::clrGet(self$ref, "SimulationBatchOptions")
```

Comment:
Could we cache `SimulationBatchOptions` as private field at `SimulationBatch` creation?